wave_vision/include/wave/vision/fast_detector.hpp

```diff
@@ -0,0 +1,106 @@
+/**
+ * @file
+ * Fast Feature Detector implementation, derived from Feature Detector base
+ * class.
+ * @ingroup vision
+ */
+#ifndef WAVE_FAST_DETECTOR_HPP
+#define WAVE_FAST_DETECTOR_HPP
+
+/** C++ Headers */
+#include <string>
+#include <vector>
+
+/** Libwave Headers */
+#include "wave/vision/feature_detector.hpp"
+
+/** The wave namespace */
+namespace wave {
+/** @addtogroup vision
+ *  @{ */
+
+/**
+ *  FASTParams struct contains the configuration parameters for the
+ *  FastFeatureDetector.
+ *
+ */
+struct FASTParams {
+    int threshold;
+    bool nonmax_suppression;
+    int type;
+};
+
+/**
+ * Representation of a feature detector using the FAST algorithm.
+ *
+ * Internally, this class is wrapping OpenCV's FastFeatureDetector module.
+ * Further reference on the FastFeatureDetector can be found
+ * [here][opencv_feature_detectors].
+ *
+ * [opencv_feature_detectors]: http://docs.opencv.org/trunk/df/d74/classcv_1_1FastFeatureDetector.html
+ */
+class FASTDetector : public FeatureDetector {
+ public:
+    /** Default Constructor */
+    FASTDetector();
+
+    /** Constructs a FastFeatureDetector using parameters specified by the user
+     *
+     *  @param config, a FASTParams struct containing the desired threshold,
```

Comment:
Nice. By the way, it is not strictly necessary to list all the @params or @returns if they are obvious.
It is never necessary to mention the type of the parameter, because it will appear in Doxygen with a hyperlink to the type's documentation.